docs/source/notes/gradcheck.rst

```diff
@@ -0,0 +1,220 @@
+2. :math:`N` and :math:`M` are two integers that we will use for the dimension of the input and output space respectively.
+
+3. :math:`f: \mathcal{R}^N \to \mathcal{R}^M` as our basic real-to-real function such that :math:`y = f(x)`.
+4. :math:`g: \mathcal{C}^N \to \mathcal{R}^M` as our basic complex to real function such that :math:`y = g(z)`.
+
+
+For the simple real-to-real case, we write as :math:`J_f` the Jacobian matrix associated with :math:`f` of size :math:`M \times N`.
+This matrix contains all the partial derivatives such that the entry at position :math:`(i, j)` contains :math:`\frac{\partial y_i}{\partial x_j}`.
+Backward mode AD is then computing, for a given vector :math:`v` of size :math:`M`, the quantity :math:`v^T J_f`.
+Forward mode AD on the other hand is computing, for a given vector :math:`u` of size :math:`N`, the quantity :math:`J_f u`.
+
+For functions that contain complex values, the story is a lot more complex. We only provide the gist here and the full description can be found at :ref:`complex_autograd-doc`.
+
+Proper complex derivatives are too restrictive for most functions we care about. So we turn to Wirtinger calculus.
+In a basic setting of Wirtinger calculus, the chain rule required access to both the Wirtinger derivative (called :math:`W` below) and the Conjugate Wirtinger derivative (called :math:`CW` below). This means that in the general case of Wirtinger calculus, the values that we would need to "backward through the graph" in backward mode AD would need to be twice the size of the "forward value" (both :math:`W` and :math:`CW` need to be propagated because in general, despite their name, one is not the complex conjugate of the other).
+
+To avoid this problem, for backward mode AD, we always work under the assumption that the current function is part of a bigger function whose output is in :math:`\mathcal{R}`. Note that this assumption means that all the intermediary gradients we compute during the backward pass are also corresponding to functions whose output is in :math:`\mathcal{R}`.
+In practice, this assumption is not restrictive when doing optimization as such problem require a real-valued objective (as there is no natural ordering of the complex numbers).
+
+Under this assumption, we can show that :math:`W = CW^*` (we use :math:`*` to denote complex conjugation here) and so only one of the two values actually need to be "backwarded through the graph" as the other one can easily be recovered.
+To simplify internal computations, PyTorch uses :math:`2 * CW` as the value it backwards and returns when the user asks for gradients.
+Similarly to the real case, when the output is actually in :math:`\mathcal{R}^M`, backward mode AD does not compute :math:`2 * CW` but only :math:`v^T (2 * CW)` for a given vector :math:`v \in \mathcal{R}^M`.
```

Comment:
Is the purpose of this to explain that we return the vector jacobian product? Regardless, all of this is explained in the note, so you could just get around writing:
`As explained in the complex autograd note <link it>, the value returned by backward mode AD is :math:`v^T * (2 * CW)` where v is a vector that belongs to the range of the function.`

Reply:
Yes the note is already linked just above. So here I just give the formula so that I can re-use it below.